### ISSUE TITLE
[Bug]: crash in internal/service/gamelift.resourceGameServerGroupCreate

### ISSUE THREAD

**Issue by abeluck** [hashicorp/terraform-provider-aws#29758]
### Terraform Core Version

1.3.8

### AWS Provider Version

4.56.0

### Affected Resource(s)

* aws_gamelift_game_server_group

### Expected Behavior

The game lift game server group should be created.

### Actual Behavior

It crashed.

### Relevant Error/Panic Output Snippet

```shell
aws_gamelift_game_server_group.this: Creating...                                                                                
╷                                                                                                                               
│ Error: Request cancelled                                                                                                      
│                                                                                                                               
│   with aws_gamelift_game_server_group.this,                                                                                   
│   on main.tf line 165, in resource "aws_gamelift_game_server_group" "this":
│  165: resource "aws_gamelift_game_server_group" "this" {                                                                      
│                                                                                                                               
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.   
╵                                                                                                                               
                                                                                                                                
Stack trace from the terraform-provider-aws_v4.56.0_x5 plugin:                                                                  
                                                                                                                                
panic: interface conversion: interface {} is nil, not string                                                                    
                                                                                                                                
goroutine 381 [running]:                                                                                                        
github.com/hashicorp/terraform-provider-aws/internal/service/gamelift.resourceGameServerGroupCreate({0xe5376c0?, 0xc00367cfc0}, 0xc003b78c80, {0xd091680?, 0xc0003cd400})
        github.com/hashicorp/terraform-provider-aws/internal/service/gamelift/game_server_group.go:243 +0x11a5                  
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xe5376c0?, {0xe5376c0?, 0xc00367cfc0?}, 0xd?, {0xd091680?, 0xc0003cd400?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:702 +0x84                                
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001376ee0, {0xe5376c0, 0xc00367cfc0}, 0xc003d22b60, 0xc003b78b00, {0xd091680, 0xc0003cd400})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:837 +0xa85                               
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00011da88, {0xe5376c0?, 0xc00367ce10?}, 0xc001d74ff0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:1021 +0xe8d                         
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc003346420, 0xc003346480, {0xc004bc8380, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0025570e0, {0xe5376c0?, 0xc00364db90?}, 0xc0005f9a40)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xcda6b00?, 0xc0025570e0}, {0xe5376c0, 0xc00364db90}, 0xc0005f99d0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0045de3c0, {0xe546640, 0xc0047876c0}, 0xc003d15440, 0xc004c62ff0, 0x14c30480, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23                                                                    
google.golang.org/grpc.(*Server).handleStream(0xc0045de3c0, {0xe546640, 0xc0047876c0}, 0xc003d15440, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f                                                                    
google.golang.org/grpc.(*Server).serveStreams.func1.2()                                                                         
        google.golang.org/grpc@v1.53.0/server.go:965 +0x98                                                                      
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.53.0/server.go:963 +0x28a                                                                     
                                                                                                                                
Error: The terraform-provider-aws_v4.56.0_x5 plugin crashed!                                                                    
                                                                                                                                
This is always indicative of a bug within the plugin. It would be immensely                                                     
helpful if you could report the crash with the plugin's maintainers so that it                                                  
can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

```terraform
resource "aws_gamelift_game_server_group" "this" {
  game_server_group_name = module.this.id

  instance_definition {
    instance_type = var.instance_type
  }

  instance_definition {
    instance_type = var.instance_type2
  }

  launch_template {
    id = aws_launch_template.this.id
  }

  max_size = 2
  min_size = 1

  role_arn    = var.role_arn
  vpc_subnets = var.gamelift_subnets
  tags        = module.this.tags
}


```

### Steps to Reproduce

1. Apply
2. Observe Crash

### Debug Output

```shell
2023-03-02T13:24:46.133Z [INFO]  provider.terraform-provider-aws_v4.56.0_x5: [INFO] Creating GameLift Game Server Group: {
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   GameServerGroupName: "REDACTED-dev-tin",
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   InstanceDefinitions: [{
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:       InstanceType: "c4.large"
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:     },{
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:       InstanceType: "m4.large"
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:     }],
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   LaunchTemplate: {
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:     LaunchTemplateId: "lt-REDACTED"
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   },
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   MaxSize: 1,
2023-03-02T13:24:46.133Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   MinSize: 2,
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   RoleArn: "arn:aws:iam::REDACTED:instance-profile/REDACTED-ec2-instance-test",
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   Tags: [{
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:       Key: "Stage",
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:       Value: "dev"
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:     },{
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:       Key: "Namespace",
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:       Value: "REDACTED"
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:     },{
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:       Key: "Name",
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:       Value: "REDACTED-dev-tin"
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:     }],
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5:   VpcSubnets: ["subnet-REDACTED","subnet-REDACTED","subnet-REDACTED"]
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: }
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: [DEBUG] Waiting for state to become: [success]
2023-03-02T13:24:46.134Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: HTTP Request Sent: aws.operation=CreateGameServerGroup aws.sdk=aws-sdk-go aws.service=GameLift http.request.header.x_amz_security_token=***** tf_provider_addr=registry.terraform.io/hashicorp/aws @module=aws @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.25/logger.go:90 http.request.header.x_amz_date=20230302T132446Z http.request_content_length=517 http.request.header.content_type=application/x-amz-json-1.1 http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.3.8 (+https://www.terraform.io) terraform-provider-aws/4.56.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.208 (go1.19.3; linux; amd64)" tf_mux_provider=*schema.GRPCProviderServer tf_resource_type=aws_gamelift_game_server_group aws.region=eu-west-1 http.flavor=1.1 http.request.body={"GameServerGroupName":"REDACTED-dev-tin","InstanceDefinitions":[{"InstanceType":"c4.large"},{"InstanceType":"m4.large"}],"LaunchTemplate":{"LaunchTemplateId":"lt-055da1e688124ab87"},"MaxSize":1,"MinSize":2,"RoleArn":"arn:aws:iam::REDACTED:instance-profile/REDACTED-ec2-instance-test","Tags":[{"Key":"Stage","Value":"dev"},{"Key":"Namespace","Value":"REDACTED"},{"Key":"Name","Value":"REDACTED-dev-tin"}],"VpcSubnets":["subnet-05e815315ee15898d","subnet-0d791a762a8f77b4e","subnet-0a167ca35ff4c0cc4"]}[truncated...] http.request.header.x_amz_target=GameLift.CreateGameServerGroup http.url=https://gamelift.eu-west-1.amazonaws.com/ net.peer.name=gamelift.eu-west-1.amazonaws.com tf_req_id=9b694c19-3be5-a041-a78b-567ffbc9d8a5 tf_rpc=ApplyResourceChange http.method=POST http.request.header.authorization="AWS4-HMAC-SHA256 Credential=ASIA************PH5E/20230302/eu-west-1/gamelift/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-security-token;x-amz-target, Signature=*****" timestamp=2023-03-02T13:24:46.133Z
/023-03-02T13:24:46.268Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: HTTP Response Received: tf_resource_type=aws_gamelift_game_server_group aws.operation=CreateGameServerGroup http.duration=134 http.response.body="{"__type":"InvalidRequestException","Message":"1 validation error detected: Value 'arn:aws:iam::REDACTED:instance-profile/REDACTED-ec2-instance-test' at 'roleArn' failed to satisfy constraint: Member must satisfy regular expression pattern: ^arn:.*:role\\/[\\w+=,.@-]+"}" http.response.header.date="Thu, 02 Mar 2023 13:24:45 GMT" http.response.header.x_amzn_requestid=e65575fe-b0fe-4e62-9416-7e64981d9cbf tf_req_id=9b694c19-3be5-a041-a78b-567ffbc9d8a5 @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.25/logger.go:138 @module=aws aws.sdk=aws-sdk-go tf_mux_provider=*schema.GRPCProviderServer tf_rpc=ApplyResourceChange http.status_code=400 tf_provider_addr=registry.terraform.io/hashicorp/aws aws.region=eu-west-1 aws.service=GameLift http.response.header.content_type=application/x-amz-json-1.1 timestamp=2023-03-02T13:24:46.268Z
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: panic: interface conversion: interface {} is nil, not string
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: goroutine 10 [running]:
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: github.com/hashicorp/terraform-provider-aws/internal/service/gamelift.resourceGameServerGroupCreate({0xe5376c0?, 0xc00539c870}, 0xc0053b0200, {0xd091680?, 0xc000402400})
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	github.com/hashicorp/terraform-provider-aws/internal/service/gamelift/game_server_group.go:243 +0x11a5
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xe5376c0?, {0xe5376c0?, 0xc00539c870?}, 0xd?, {0xd091680?, 0xc000402400?})
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:702 +0x84
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001443b20, {0xe5376c0, 0xc00539c870}, 0xc005396dd0, 0xc0053b0080, {0xd091680, 0xc000402400})
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:837 +0xa85
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002050e0, {0xe5376c0?, 0xc00539c750?}, 0xc000137e00)
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:1021 +0xe8d
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc003342990, 0xc0033429f0, {0xc004e98380, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc002830e60, {0xe5376c0?, 0xc00538fa10?}, 0xc003be9960)
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xcda6b00?, 0xc002830e60}, {0xe5376c0, 0xc00538fa10}, 0xc003be98f0, 0x0)
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
2023-03-02T13:24:46.270Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00466de00, {0xe546640, 0xc000240680}, 0xc005394900, 0xc004f30cc0, 0x14c30480, 0x0)
2023-03-02T13:24:46.271Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23
2023-03-02T13:24:46.271Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc00466de00, {0xe546640, 0xc000240680}, 0xc005394900, 0x0)
2023-03-02T13:24:46.271Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f
2023-03-02T13:24:46.271Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.2()
2023-03-02T13:24:46.271Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	google.golang.org/grpc@v1.53.0/server.go:965 +0x98
2023-03-02T13:24:46.271Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2023-03-02T13:24:46.271Z [DEBUG] provider.terraform-provider-aws_v4.56.0_x5: 	google.golang.org/grpc@v1.53.0/server.go:963 +0x28a
2023-03-02T13:24:46.284Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.56.0/linux_amd64/terraform-provider-aws_v4.56.0_x5 pid=64435 error="exit status 2"
2023-03-02T13:24:46.284Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-03-02T13:24:46.284Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-03-02T13:24:46.290Z [ERROR] vertex "aws_gamelift_game_server_group.this" error: Plugin did not respond
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with aws_gamelift_game_server_group.this,
[31m│[0m [0m  on main.tf line 165, in resource "aws_gamelift_game_server_group" "this":
[31m│[0m [0m 165: resource "aws_gamelift_game_server_group" "this" [4m{[0m[0m
[31m│[0m [0m
[31m│[0m [0mThe plugin encountered an error, and failed to respond to the
[31m│[0m [0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
[31m│[0m [0mcontain more details.
[31m╵[0m[0m

Stack trace from the terraform-provider-aws_v4.56.0_x5 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 10 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/gamelift.resourceGameServerGroupCreate({0xe5376c0?, 0xc00539c870}, 0xc0053b0200, {0xd091680?, 0xc000402400})
	github.com/hashicorp/terraform-provider-aws/internal/service/gamelift/game_server_group.go:243 +0x11a5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xe5376c0?, {0xe5376c0?, 0xc00539c870?}, 0xd?, {0xd091680?, 0xc000402400?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:702 +0x84
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001443b20, {0xe5376c0, 0xc00539c870}, 0xc005396dd0, 0xc0053b0080, {0xd091680, 0xc000402400})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002050e0, {0xe5376c0?, 0xc00539c750?}, 0xc000137e00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc003342990, 0xc0033429f0, {0xc004e98380, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc002830e60, {0xe5376c0?, 0xc00538fa10?}, 0xc003be9960)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xcda6b00?, 0xc002830e60}, {0xe5376c0, 0xc00538fa10}, 0xc003be98f0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00466de00, {0xe546640, 0xc000240680}, 0xc005394900, 0xc004f30cc0, 0x14c30480, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc00466de00, {0xe546640, 0xc000240680}, 0xc005394900, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-aws_v4.56.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2023-03-02T13:24:46.298Z [DEBUG] provider: plugin exited
```

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

No

**Comment by abeluck** [comment]
The crash is happening in the error handler.. obscuring the real error.

**Comment by thomasrost99** [comment]
Was having the same issue so I looked into debug logs. This is very likely a case of the game server group name needing to follow a specific regex pattern. I think this just needs to be handled better in the provider.
`validation error detected: Value 'game_server_group' at 'gameServerGroupName' failed to satisfy constraint: Member must satisfy regular expression pattern: [a-zA-Z0-9-\\.]+"}`
For me, I was using underscores, but until a fix is pushed for this I would **double check naming conventions** or set your log level to debug and look into those
tf version 1.7.3
aws provider version 5.38.0

**Comment by abeluck** [comment]
@justinretzolk It's been over a year and this plugin crash still exists. Does it still need triaging?

**Comment by jar-b** [comment]
Fixed with b2e36ed1979a710d73b4ddde2bceb9f4a93d53af (#38379).

The create method error return incorrectly referenced a `name` argument (which does not exist in the schema) and attempted to insert the resulting `nil` value to a string. With the changes above, the return now references the correct `game_server_group_name` argument.

```go
	if err != nil {
		return sdkdiag.AppendErrorf(diags, "creating GameLift Game Server Group (%s): %s", d.Get("name").(string), err)
	}
```
